src/button/button-svg.c

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Copyright (C) Johan Malm 2023
+ */
+#define _POSIX_C_SOURCE 200809L
+#include <cairo.h>
+#include <librsvg/rsvg.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <wlr/util/log.h>
+#include "buffer.h"
+#include "button/button-svg.h"
+#include "button/common.h"
+#include "common/file-helpers.h"
+#include "labwc.h"
+#include "theme.h"
+
+void
+button_svg_load(const char *button_name, struct lab_data_buffer **buffer,
+		int size)
+{
+	if (*buffer) {
+		wlr_buffer_drop(&(*buffer)->base);
+		*buffer = NULL;
+	}
+
+	char *filename = button_filename(button_name);
+	if (!file_exists(filename)) {
+		return;
+	}
+
+	GError *err = NULL;
+	RsvgRectangle viewport = { .width = size, .height = size };
+	RsvgHandle *svg = rsvg_handle_new_from_file(filename, &err);
+	if (err) {
+		wlr_log(WLR_ERROR, "error reading svg %s-%s\n", filename, err->message);
+		g_error_free(err);
+		return;
+	}
+
+	cairo_surface_t *image = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, size, size);
+
+	cairo_t *cr = cairo_create(image);
+	rsvg_handle_render_document(svg, cr, &viewport, &err);
+	if (err) {
+		wlr_log(WLR_ERROR, "error rendering svg %s-%s\n", filename, err->message);
+		g_error_free(err);
+		return;
+	}
```

Comment:
No need to free cairo stuff in the err condition?

Reply:
Good catch. Done.